VK/VK/ShowGriendPhotoScreen/ShowFriendPhotoScreenViewController.swift

```diff
@@ -0,0 +1,169 @@
+            imageNames.append(imageName)
+        }
+
+        for image in imageNames {
+            guard let slide: ShowPhotoView = Bundle.main.loadNibNamed(
+                NibNames.showPhotoViewNibNameText,
+                owner: self,
+                options: nil
+            )?
+                .first as? ShowPhotoView else { return [] }
+            slide.photoImageView.image = UIImage(named: image)
+            let recognizer = UISwipeGestureRecognizer(target: self, action: #selector(gestoreAction))
+            recognizer.direction = .down
+            slide.photoImageView.addGestureRecognizer(recognizer)
+            slide.photoImageView.isUserInteractionEnabled = true
+            slides.append(slide)
+        }
+
+        return slides
+    }
+
+    private func configureScrollView() {
+        guard let slidesForScroll = slides else { return }
+        showPhotoScrollView.frame = CGRect(x: 0, y: 0, width: view.frame.width, height: view.frame.height)
+        showPhotoScrollView.contentSize = CGSize(
+            width: view.frame.width * CGFloat(slidesForScroll.count),
+            height: view.frame.height
+        )
+        showPhotoScrollView.isPagingEnabled = true
+
+        for index in 0 ..< slidesForScroll.count {
+            slidesForScroll[index]
+                .frame = CGRect(
+                    x: view.frame.width * CGFloat(index),
+                    y: 0,
+                    width: view.frame.width,
+                    height: view.frame.height
+                )
+            showPhotoScrollView.addSubview(slidesForScroll[index])
+        }
+    }
+
+    private func configurePageControlView() {
+        pageControllView.numberOfPages = slides?.count ?? 0
+        pageControllView.currentPage = 0
+        view.bringSubviewToFront(pageControllView)
+    }
+
+    private func configureTapHandler() {
+        tapHandler = {
+            self.dismiss(animated: true)
+        }
+    }
+
+    @objc private func gestoreAction() {
+        guard let action = tapHandler else { return }
+        action()
+    }
+}
+
+// MARK: - UIScrollViewDelegate
+
+extension ShowFriendPhotoScreenViewController: UIScrollViewDelegate {
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        guard let slides = slides else { return }
+        let pageIndex = round(scrollView.contentOffset.x / view.frame.width)
+        pageControllView.currentPage = Int(pageIndex)
+
+        let maximumHorizontalOffset: CGFloat = scrollView.contentSize.width - scrollView.frame.width
+        let currentHorizontalOffset: CGFloat = scrollView.contentOffset.x
+
+        let maximumVerticalOffset: CGFloat = scrollView.contentSize.height - scrollView.frame.height
+        let currentVerticalOffset: CGFloat = scrollView.contentOffset.y
+
+        let percentageHorizontalOffset: CGFloat = currentHorizontalOffset / maximumHorizontalOffset
+        let percentageVerticalOffset: CGFloat = currentVerticalOffset / maximumVerticalOffset
```

Comment:
```suggestion
        let percentageHorizontalOffset: CGFloat = currentHorizontalOffset / maximumHorizontalOffset
        let percentageVerticalOffset: CGFloat = currentVerticalOffset / maximumVerticalOffset
```
можно поймать деление на 0